x-pack/plugins/observability/public/components/shared/data_picker/index.tsx

```diff
@@ -37,17 +38,31 @@ export function DatePicker({ rangeFrom, rangeTo, refreshPaused, refreshInterval
   const history = useHistory();
   const { plugins } = usePluginContext();
 
-  useEffect(() => {
-    plugins.data.query.timefilter.timefilter.setTime({
-      from: rangeFrom,
-      to: rangeTo,
-    });
-  }, [plugins, rangeFrom, rangeTo]);
-
   const timePickerQuickRanges = useKibanaUISettings<TimePickerQuickRange[]>(
     UI_SETTINGS.TIMEPICKER_QUICK_RANGES
   );
 
+  const timeRange = useTimeRange({ rangeFrom, rangeTo });
+
+  useEffect(() => {
+    if (rangeFrom && rangeTo) {
+      plugins.data.query.timefilter.timefilter.setTime({
+        from: rangeFrom,
+        to: rangeTo,
+      });
+      return;
+    }
+
+    history.replace({
+      ...location,
+      search: fromQuery({
+        ...toQuery(location.search),
+        rangeFrom: timeRange.rangeFrom,
+        rangeTo: timeRange.rangeTo,
+      }),
+    });
+  }, [plugins, location, history, timeRange, rangeFrom, rangeTo]);
```

Comment:
This looks like a pretty eager hook, in the sense that it has a couple of dependencies that are not simple primitives (e.g., plugins & timeRange). I have some concerns around it aggressively calling `history.replace`. Should we be more explicit with our dependencies?